.github/workflows/regression.yml

```diff
@@ -19,7 +22,15 @@ jobs:
         python-version: ['3.7', '3.8', '3.9', '3.10']
 
     steps:
-    - uses: actions/checkout@v3
+    - name: Run actions/checkout@v3
+      if: ${{ github.event_name != 'schedule' }}
+      uses: actions/checkout@v3
+
+    - name: Run nightly regression test on master branch
+      if: ${{ github.event_name == 'schedule' }}
+      uses: actions/checkout@v3
+      with:
+        ref: master
```

Comment:
okay how about let's focus on testing the `devel` branch of the code on nightly basis. I particularly dont like the fact we have this `if` condition.

Reply:
@shahzebsiddiqui Okay done. Just out of curiosity, what issues do you think the `if` condition can cause ?

Reply:
yeah i think the workflow could get hard to troubleshoot considering there are now 2 more steps and only one step will run. Originally i was envisioning a new workflow file (`.yml`) but this works better.

Reply:
the only benefit of a new workflow file is that we can get a new badge and be able to see the nightly builds. Right now this badge is used for this workflow which typically corresponds to the check run from the PR
![Screen Shot 2023-02-10 at 9 57 02 AM](https://user-images.githubusercontent.com/12942230/218122931-2dce66b7-d255-4563-bf1d-7186637fac13.png)